Bug 1223890 - Constraint errors in history panel

Johennes opened the pull request.

thebnich commented on Client/Frontend/Widgets/SiteTableViewController.swift:
```diff
@@ -52,9 +52,13 @@ class SiteTableViewHeader : UITableViewHeaderFooterView {
         titleLabel.snp_makeConstraints { make in
-            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin)
-            make.right.equalTo(contentView).offset(-SiteTableViewControllerUX.HeaderTextMargin)
+            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin).priority(999)
```
SnapKit includes a `priorityHigh()` -- I think that'd be slightly cleaner than manually defining a priority here.

thebnich commented on Client/Frontend/Widgets/SiteTableViewController.swift:
```diff
@@ -52,9 +52,13 @@ class SiteTableViewHeader : UITableViewHeaderFooterView {
-            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin)
-            make.right.equalTo(contentView).offset(-SiteTableViewControllerUX.HeaderTextMargin)
+            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin).priority(999)
+            make.left.greaterThanOrEqualTo(contentView)
```
Are these `greaterThanOrEqualTo` and `lessThanOrEqualTo` lines necessary?

Johennes commented:
I think that technically the `greaterThanOrEqualTo` and `lessThanOrEqualTo` constraints are required. When the view is initialized with a zero size, the left and right space constraints will break due to their non-required priority. That's what we want to not impose a minimum size on the view that would conflict with the initial zero size and result in the warning on the console. However, when these constraints break, it also means that the label's x position is no longer determined, unless we add any kind of fallback constraints.

So that's the theory. Practically, however, I could not see any issue on the console when omitting the `greaterThanOrEqualTo` and `lessThanOrEqualTo` fallback constraints. I thought that it'll still be safer to add them but if you think they're superfluous or throw up too many questions, I can remove them. Just let me know.

Other than that, I updated the PR and replaced the `999` literal with `priorityHigh`. I guess the `750` is still high enough for our purposes.

Johennes commented:
@thebnich Anything else I should change? Shall I rather remove the `greaterThanOrEqualTo` / `lessThanOrEqualTo` fallback constraints.

thebnich commented:
Works for me. Thanks!

Merged 889206b.
